src/envoy/utils/http_header_utils.cc

```diff
@@ -38,13 +38,17 @@ absl::string_view extractHeader(const Envoy::Http::HeaderMap& headers,
   return readHeaderEntry(entry);
 }
 
-absl::string_view getRequestHTTPMethodWithOverride(
-    absl::string_view originalMethod, const Envoy::Http::HeaderMap& headers) {
+bool handleHttpMethodOverride(Envoy::Http::HeaderMap& headers) {
   const auto* entry = headers.get(kHttpMethodOverrideHeader);
   if (entry) {
-    return entry->value().getStringView();
+    // Move the header.
+    headers.remove(Envoy::Http::Headers::get().Method);
+    headers.addCopy(Envoy::Http::Headers::get().Method,
+                    entry->value().getStringView());
```

Comment:
you may be using a freed memory after removing Method.

Reply:
I think `HeaderMap` was recently re-written to not have this problem. ASan should catch this as well, as we have c++ unit tests that cover this.